controllers/submariner/cleanup.go

```diff
@@ -39,14 +39,21 @@ import (
 var requeueAfter = reconcile.Result{RequeueAfter: time.Millisecond * 100}
 
 func (r *Reconciler) runComponentCleanup(ctx context.Context, instance *operatorv1alpha1.Submariner) (reconcile.Result, error) {
+	objsToDelete := []client.Object{
+		newDaemonSet(names.GatewayComponent, instance.Namespace),
+		newDaemonSet(names.RouteAgentComponent, instance.Namespace),
+	}
+
 	// First, delete the regular DaemonSets/Deployments and ensure all their pods are cleaned up.
-	objsToDelete := []client.Object{newDaemonSet(names.GatewayComponent, instance.Namespace)}
 	for _, obj := range objsToDelete {
 		err := r.ensureDeleted(ctx, obj)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
+	}
 
+	// Next, delete ensure all their pods are cleaned up.
```

Comment:
// Next, delete `and` ensure all their pods are cleaned up.?

Reply:
yeah that's fixed in a subsequent commit.